### PR TITLE
FOGL-9307 Refactor traceCallback to Utilize MQTT Trace Levels for Logging

### PR DESCRIPTION
**Pull request by ashwini-k-pandey**
**Summary**:
This pull request refactors the traceCallback function to improve logging behavior based on the different trace levels defined in the MQTT Client library. The changes enhance the clarity and usefulness of log messages within the FogLAMP logging system.

### CODE REVIEW

File: scripted.cpp [hunk 1]
```diff
@@ -175,7 +175,34 @@ void MQTTScripted::processPolicy(const string& policy)
...
+        case MQTTCLIENT_TRACE_PROTOCOL:
+            Logger::getLogger()->debug("Protocol Trace: %s", message);
+            break;
+
+        case MQTTCLIENT_TRACE_ERROR:
+            Logger::getLogger()->error("Error Trace: %s", message);
+            break;
+
+        case MQTTCLIENT_TRACE_SEVERE:
+            Logger::getLogger()->fatal("Severe Trace: %s", message);
```
Comment (gnandan) [inline scripted.cpp:195]: Should it not be logged as error instead of fatal ?

Comment (ashwini-k-pandey) [reply scripted.cpp:195]: The MQTTCLIENT library does not provide a standardized definition for log levels, making it challenging to align its trace levels with our logging framework. To ensure robust error handling and prevent potential loss of critical information, I opted for a conservative approach by mapping the 'Severe' trace level to 'Fatal' in our logging system.

Comment (praveen-garg) [reply scripted.cpp:195]: If fatal will halt then we should keep error.

Comment (ashwini-k-pandey) [reply scripted.cpp:195]: Logger::fatal(const string& msg, ...) --> This code logs syslg with log level LOG_CRIT. Fatal do not halt execution

**Review by gnandan — APPROVED** [review]

**Pull request merged**